scripts/tests/test_fsx.sh

```
@@ -0,0 +1,152 @@
+  echo "Building fsx (release) ..."
+  # Override any host CARGO_TARGET_DIR so the binary always lands in a
+  # predictable spot under FSX_DIR — otherwise we have to chase wherever
+  # the env happened to point cargo.
+  CARGO_TARGET_DIR="$FSX_DIR/build" \
+    cargo build --release --quiet --manifest-path "$FSX_DIR/Cargo.toml"
+  FSX_BIN="$FSX_DIR/build/release/fsx"
+  test -x "$FSX_BIN" || { echo "FAIL: fsx binary not built"; exit 1; }
+}
+
+create_seed_archive() {
+  ( cd "$WORKDIR" && echo "seed" > seed.txt && \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)